fedimint-client/src/lib.rs

```diff
@@ -1258,21 +1258,77 @@ impl TransactionUpdates {
     }
 }
 
+/// Federation config and meta data that can be used to show a preview of the
+/// federation one is about to join or to initialize a client.
+#[derive(Debug, Clone)]
+pub struct FederationInfo {
+    config: ClientConfig,
+    // TODO: make non-optional or remove
+    invite_code: Option<InviteCode>,
+}
+
+impl FederationInfo {
+    /// Download federation info using invitation code
+    pub async fn from_invite_code(invite: InviteCode) -> anyhow::Result<FederationInfo> {
+        let config = try_download_config(invite.clone(), 10).await?;
+        Ok(FederationInfo {
+            config,
+            invite_code: Some(invite),
+        })
+    }
+
+    /// Create `FederationInfo` from config, may download further meta data in
+    /// the future
+    pub async fn from_config(config: ClientConfig) -> anyhow::Result<FederationInfo> {
+        // The return type is a result in case we want to fallibly fetch additional meta
+        // data in the future
+        Ok(FederationInfo {
+            config,
+            invite_code: None,
+        })
+    }
+
+    /// Returns the federations configuration
+    pub fn config(&self) -> &ClientConfig {
+        &self.config
+    }
```

Comment:
Do we actually want to make that available or should there be functions for all the relevant fields? @wbobeirne @justinmoon

Reply:
I think just spitting up the whole config is best for now